Lib/ctypes/__init__.py

```diff
@@ -266,6 +266,28 @@ class c_bool(_SimpleCData):
 
 from _ctypes import POINTER, pointer, _pointer_type_cache
 
+_CType_Type = type(_Pointer).__base__
+
+def POINTER(cls):
+    if cls is None:
+        return c_void_p
+    try:
+        pt = cls.__pointer_type__
+        if pt is not None:
+            return pt
+    except AttributeError:
+        pass
+    if isinstance(cls, str):
+        return type(f'LP_{cls}', (_Pointer,), {})
+    if isinstance(cls, _CType_Type):
+        return type(f'LP_{cls.__name__}', (_Pointer,), {'_type_': cls})
+
+    raise TypeError(f'must be a ctypes-like type: {cls}')
+
```

Comment:
Do you think `PyCPointerType_SetProto()` has sufficient type checks?

```suggestion
def POINTER(cls):
    if cls is None:
        return c_void_p
    try:
        pt = cls.__pointer_type__
        if pt is not None:
            return pt
    except AttributeError:
        pass
    if isinstance(cls, str):
        return type(f'LP_{cls}', (_Pointer,), {})
    return type(f'LP_{cls.__name__}', (_Pointer,), {'_type_': cls})
```

Reply:
Thanks!
I have added extra checks. I believe it is sufficient now. Also, I removed 'deprecated' comment, cause I think this case is not deprecated.